### PR TITLE
Gist integration

### PR DESCRIPTION
**Pull request by mikedewar**
Two SUPER basic endpoints for exporting and importing groups as gists. Designed really to start exploring the space, so aren't particularly fancy, but do the job. Uses the settings file so maybe look at the settings file PR first.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -27,8 +27,23 @@ const (
 	PARAM = "param"
 )
 
+// user-session specific settings
+type Settings struct {
+	Autosave        bool
```
Comment (nikhan) [inline server/server.go]: as far as i can tell, this is never used.

File: server/server.go [hunk 2]
```diff
@@ -27,8 +27,23 @@ const (
...
+	GithubUserToken string
+}
+
+// NewSettings returns the default settings object
+func NewSettings() Settings {
+	return Settings{
+		Autosave:        true,
```
Comment (nikhan) [inline server/server.go]: neither is this.

**Comment by mikedewar** [comment]
removed references to Autosave (which were there just to test out the settings file, before the user token setting)

**Comment by nikhan** [comment]
squash pls

**Comment by mikedewar** [comment]
done...

**Comment by nikhan** [comment]
:dog: :dog:

**Pull request merged**